### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by yogeshjain999**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,62 @@
...
+
+_Dynamic options to evaluate at runtime._
+
+`Trailblazer::Option` is the one of the core concept behind `traiblazer-operation`'s [step API](https://trailblazer.to/2.1/docs/activity.html#activity-wiring-api), `reform`'s [populator API](https://trailblazer.to/2.1/docs/reform.html#reform-populators) etc. It makes us possible to accept any kind of callable objects at compile time and execute them at runtime.
+
+```ruby
+class Song::Create < Trailblazer::Operation
+  step Authorize				# Module callable
+  step :model					# Method callable
+  step ->(ctx, model:, **ctx) { puts model }	# Proc callable
```
Comment (seuros) [inline README.md]: both params named the same ?

Comment (yogeshjain999) [reply README.md]: ah my bad, updated it.

**Review by seuros — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1 +1,62 @@
...
+  def self.call(*args, message:, **options)
+    message
+  end
+end
+
+option = Trailblazer::Option(CallMe)
+option.(*args, keyword_arguments: { message: "hello!" }, exec_context: nil) => "hello!"
+```
+
+Notice the usage of `keyword_arguments` while calling an `Option`. This is because keyword arguments needs to be forwarded separately in order for them to be [compatible](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/) with ruby >= 2.7.
```
Comment (seuros) [inline README.md]: with ruby 2.7+

**Review by seuros — APPROVED** [review]

**Review by apotonick — APPROVED** [review]
This is great! Can we have that on the website, too? :beers:

**Comment by yogeshjain999** [comment]
> This is great! Can we have that on the website, too? 🍻

👍  I was going to do that once this gets merged.

**Pull request merged**